chia/rpc/wallet_rpc_api.py

```diff
@@ -378,39 +392,42 @@ async def get_latest_singleton_coin_spend(
     # Key management
     ##########################################################################################
 
-    async def log_in(self, request: Dict[str, Any]) -> EndpointResult:
+    @marshal
+    async def log_in(self, request: LogIn) -> LogInResponse:
         """
         Logs in the wallet with a specific key.
         """
 
-        fingerprint = request["fingerprint"]
-        if self.service.logged_in_fingerprint == fingerprint:
-            return {"fingerprint": fingerprint}
+        if self.service.logged_in_fingerprint == request.fingerprint:
+            return LogInResponse(request.fingerprint)
 
         await self._stop_wallet()
-        started = await self.service._start_with_fingerprint(fingerprint)
+        started = await self.service._start_with_fingerprint(request.fingerprint)
         if started is True:
-            return {"fingerprint": fingerprint}
+            return LogInResponse(request.fingerprint)
 
-        return {"success": False, "error": f"fingerprint {fingerprint} not found in keychain or keychain is empty"}
+        raise ValueError(f"fingerprint {request.fingerprint} not found in keychain or keychain is empty")
 
-    async def get_logged_in_fingerprint(self, request: Dict[str, Any]) -> EndpointResult:
-        return {"fingerprint": self.service.logged_in_fingerprint}
+    @marshal
+    async def get_logged_in_fingerprint(self, request: Empty) -> GetLoggedInFingerprintResponse:
```

Comment:
How did you settle on this required-`Empty` solution vs. `Empty` with a default or just removing the parameter?  I have a few immediate thoughts, but curious to just hear yours as I'm guessing you spent more than seconds considering this.  `:]`

Reply:
I guess I didn't look that hard into it, but I'm pretty sure the `@marshal` decorator (or maybe the `RpcServer` class above it) forces the request parameter in there.  I just took that for granted and decided it had to be there so I decided on the `Empty` class as a hint.

Reply:
I think I was thinking on the client side but commented here...  `:[`  but the client side doesn't have a "pointless" parameter, so I'll just retract this thought entirely.